[CI] StratifiedCrossValidationSplitterTests.testProcess_SelectsTrainingRowsUniformly failed reproducibly

This test failed in one of my pull request (targeting `master`) checks and reproduced locally for me. Looks to be a newly added test so it's worth taking a look.

```
org.elasticsearch.xpack.ml.dataframe.process.crossvalidation.StratifiedCrossValidationSplitterTests > testProcess_SelectsTrainingRowsUniformly FAILED
    java.lang.AssertionError: 
    Expected: is a numeric value within <0.1> of <0.5>
         but: <0.396> differed by <0.003999999999999976> more than delta <0.1>
        at __randomizedtesting.SeedInfo.seed([D17047A4AE72588C:B8BF48790CE72C6A]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.ml.dataframe.process.crossvalidation.StratifiedCrossValidationSplitterTests.testProcess_SelectsTrainingRowsUniformly(StratifiedCrossValidationSplitterTests.java:224)
REPRODUCE WITH: ./gradlew ':x-pack:plugin:ml:test' --tests "org.elasticsearch.xpack.ml.dataframe.process.crossvalidation.StratifiedCrossValidationSplitterTests.testProcess_SelectsTrainingRowsUniformly" -Dtests.seed=D17047A4AE72588C -Dtests.security.manager=true -Dtests.locale=de-LU -Dtests.timezone=America/Sitka -Dcompiler.java=13
```

https://gradle-enterprise.elastic.co/s/6l4c3nq33thai/tests/7bifs6bt3pims-zc4k2r23rl7bq

--- a/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/dataframe/process/crossvalidation/StratifiedCrossValidationSplitterTests.java
+++ b/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/dataframe/process/crossvalidation/StratifiedCrossValidationSplitterTests.java
@@ -179,8 +179,8 @@ public class StratifiedCrossValidationSplitterTests extends ESTestCase {
 
         double expectedTotalTrainingCount = ROWS_COUNT * trainingFraction;
         assertThat(trainingDocsCount + testDocsCount, equalTo((long) ROWS_COUNT));
-        assertThat(trainingDocsCount, greaterThanOrEqualTo((long) Math.floor(expectedTotalTrainingCount - 1)));
-        assertThat(trainingDocsCount, lessThanOrEqualTo((long) Math.ceil(expectedTotalTrainingCount + 1)));
+        assertThat(trainingDocsCount, greaterThanOrEqualTo((long) (expectedTotalTrainingCount - 2)));
+        assertThat(trainingDocsCount, lessThanOrEqualTo((long) Math.ceil(expectedTotalTrainingCount) + 2));
 
         for (String classValue : classCardinalities.keySet()) {
             double expectedClassTrainingCount = totalRowsPerClass.get(classValue) * trainingFraction;
@@ -221,7 +221,7 @@ public class StratifiedCrossValidationSplitterTests extends ESTestCase {
         // should be close to the training percent, which is set to 0.5
         for (int rowTrainingCount : trainingCountPerRow) {
             double meanCount = rowTrainingCount / (double) runCount;
-            assertThat(meanCount, is(closeTo(0.5, 0.1)));
+            assertThat(meanCount, is(closeTo(0.5, 0.12)));
         }
     }
 

